Balancer-Bots/Large-Internal-Balance-Deposit-Withdrawal/src/balance.fetcher.ts

```diff
@@ -5,12 +5,16 @@ import { TOKEN_ABI } from "./constants";
 
 export default class BalanceFetcher {
   provider: providers.Provider;
-  private cache: LRU<string, BigNumber>;
+  private balanceCache: LRU<string, BigNumber>;
+  private symbolCache: LRU<string, string>;
   private tokenContract: Contract;
 
   constructor(provider: providers.Provider) {
     this.provider = provider;
-    this.cache = new LRU<string, BigNumber>({
+    this.balanceCache = new LRU<string, BigNumber>({
+      max: 10000,
+    });
+    this.symbolCache = new LRU<string, string>({
       max: 10000,
     });
```

Comment:
Just a comment, that you could have the same cache for both by using different keys.